src/Diagnostics.RuntimeHost/Services/SourceWatcher/Workers/GithubWorkerBase.cs

```diff
@@ -22,6 +25,15 @@ public abstract class GithubWorkerBase : IGithubWorker
         /// </summary>
         public abstract string Name { get; }
 
+        private readonly bool _loadOnlyPublicDetectors;
+
+        public GithubWorkerBase(bool loadOnlyPublicDetectors)
+        {
+            _loadOnlyPublicDetectors = loadOnlyPublicDetectors;
+        }
+
+        private static Regex regexPublicDetectors = new Regex(@"InternalOnly\s*=\s*false",RegexOptions.IgnoreCase);
```

Comment:
Still a possibility with this RegEx to catch few internal detectors.

Reply:
Catching a few internal detectors shouldn't slow things down too much, I would prefer to be too permissive then accidentally exclude some external detector.